exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillDatabaseMetaDataImpl.java

```diff
@@ -184,6 +184,17 @@ private ServerMeta getServerMeta() throws SQLException {
     return serverMeta;
   }
 
+  /**
+   * The same as {@link DrillDatabaseMetaDataImpl#getServerMeta()} but updates server metadata for the every call
+   * @return server meta information
+   * @throws SQLException for error when getting server meta
+   */
+  private ServerMeta getUpdatedServerMeta() throws SQLException {
```

Comment:
You are right regarding SQLServer JDBC metadata.
Eventually I think for now we can left `getIdentifierQuoteString()` without updating for every call until it will be really necessary to handle direct session changes in DatabaseMetadata.